example/withoutSuspense/Firestore.tsx

```diff
@@ -113,22 +112,13 @@ const StaticValue = () => {
 };
 
 export const Firestore = () => {
-  const firebaseApp = useFirebaseApp();
-  const [firestoreInstance, setFirestoreInstance] = useState<FirebaseFirestore | undefined>();
-
-  // idea: useFirestoreInstance();
-  // this could set up enableIndexedDbPersistence
-  // and emulators!
-  // https://github.com/angular/angularfire/issues/1158
-
-  useEffect(() => {
+  const {status, data: firestoreInstance} = useInitFirestore(async (firebaseApp) => {
```

Comment:
This looks really good for custom initialization. If you want to support lazy-loading, we'll need to have an API that stores the entire module so it's easier to retrieve later without having to async fetch it again.

```ts
const { status, data: firestoreInstance } = useInitFirestore(
    async firebaseApp => {
       // It does this behind this scenes -> import('firebase/firestore');
       // and then sets to a provider so you can import it later.
      const firestoreModule = await useLazyFirestore();
      const db = firestoreModule.initializeFirestore(firebaseApp, { });
      await firestoreModule.enableIndexedDbPersistence(db);
      return db;
    }
  );
```

That' another API for another day. But it's good to consider it for this design.